src/__tests__/encode-test.js

```diff
@@ -17,6 +17,7 @@ const mockObject = function(className) {
 };
 mockObject.registerSubclass = function() {};
 mockObject.prototype = {
+  id : 'objId123',
```

Comment:
when i remove this line `id` is `undefined` and the new introduced `if (value.id === undefined)` matches and the test `encode ParseObjects` fails because the test expects a pointer. But this is exactly the case I want to treat differently. When there is no `id` toPointer should not be called and the object should be send as full json.

```
  ● encode › encodes ParseObjects
    Expected: "POINTER"
    Received: {"__type": "Object", "className": "Item"}
```

Reply:
Update the test by setting an id within the test. The MockObject should closely resemble a real ParseObject.